client/extensions/woocommerce/state/sites/shipping-classes/actions.js

```diff
@@ -19,10 +20,31 @@ export const fetchShippingClassesSuccess = ( siteId, data ) => {
 	};
 };
 
-export const fetchShippingClassesFailure = () => {
-	return errorNotice(
-		`Could not retrieve the shipping classes for this website. Please refresh and try again!`
-	);
+export const fetchShippingClassesFailure = ( action, error, dispatch ) => {
+	const { siteId } = action;
+	let noticeAction = null;
+
+	const onRetryClick = e => {
+		const {
+			notice: { noticeId },
+		} = noticeAction;
```

Comment:
I think you can avoid this variable if you pass an `id` to the `errorNotice()` options.

Reply:
I hadn't taken a look at the generation of IDs within `errorNotice`/`createNotice` and thought it would be something more complicated, but seeing that it's just about calling `uniqueId`, I switched to a local constant and it looks cleaner now.